.clang-tidy

```diff
@@ -0,0 +1,8 @@
+---
```

Comment:
non engine related changes must go as separate PR. also I saw open one already

Reply:
Yeah - will rebase this once i finish it. But for now I would like to have the `clang-tidy` warnings in my ide as well as the `create_project` patch (which is also a separate pr already). But leaving this here as a reminder

Reply:
Added to the todo list at the pr description - so closing the reminder here.